scarlet2/initialization.py

```diff
@@ -0,0 +1,304 @@
+            cutout_obs = cut_square_box(band, center, box[0])
+            cutout_noise = cut_square_box(noise_rms[idx], center, box[0])
+            perimeter_pixels[idx] = jnp.array([perimeter_values(cutout_obs)])
+            perimeter_noise_rms[idx] = jnp.array([perimeter_values(cutout_noise)])
+        perimeter_pixels = jnp.squeeze(jnp.stack(perimeter_pixels, axis=0), axis=1)
+        perimeter_noise_rms = jnp.squeeze(
+            jnp.stack(perimeter_noise_rms, axis=0), axis=1
+        )
+
+        # simplified spectrum calculation for perimeter average around cutout
+        models = (None,) * len(observation.data)
+        spectra = []
+        spectrum_avg = 0
+        for obs, model in zip((observation,), models):
+            psf_model = obs.frame.psf()
+            psf_peak = psf_model.max(axis=(1, 2))
+            for i in range(perimeter_pixels.shape[1]):
+                spectrum = perimeter_pixels[:, i]
+                spectrum /= psf_peak
+                spectrum_avg += spectrum
+            spectrum_avg /= perimeter_pixels.shape[1]
+            spectra.append(spectrum_avg)
+        spectra_box = spectra[0]
+
+        # now check the SNR and spectrum correlations
+        noise_avg = jnp.mean(perimeter_noise_rms, axis=1) * noise_thresh
+        snr = spectra_box / noise_avg
+        spec_corr = jnp.dot(spectra_box, peak_spec) / jnp.dot(
+            jnp.sqrt(jnp.dot(spectra_box, spectra_box)),
+            jnp.sqrt(jnp.dot(peak_spec, peak_spec)),
+        )
+
+        # check if SNR is below threshold or spectrum correlation is below threshold
+        if snr.all() < 1 or spec_corr < corr_thresh:
+            converged = True
+        # check if flux on boundaries increased with larger size
+        elif (spectra_box >= peak_spec).any() or (spectra_box >= spectra_prev).any():
+            converged = True
+        # increase box/morphology size
+        else:
+            spectra_prev = spectra_box
+            size += 5
+        # for debugging
+        if sigma > 5:
+            break
+
+    # normalise the morphology between 0 and 1
+    #morph = morph()
+    morph = (morph - jnp.min(morph)) / (jnp.max(morph) - jnp.min(morph))
+    return morph
+
+
+def init_morphology(
+    obs,
+    center,
+    psf_sigma=1,
+    noise_thresh=100,
+    corr_thresh=0.8,
+    max_size=32,
+    components=1,
+):
+    """Initialize the morphology of the sources.
+    Parameters
+    ----------
+    obs : `~scarlet.observation.Observation` instance
+    center : tuple (int, int) central pixel of the sources
+
+    Returns
+    -------
+    morph : np.ndarray morphology of the source normalised to (0,1)
+    """
+    morph = init_simple_morph(obs, center, psf_sigma, noise_thresh, corr_thresh)
+    if (morph.shape[0] <= max_size) or (components == 1):
+        return morph
+
+    # fit for a more complex morphology for bigger sources
+    else:
+        bx = morph.shape[0]
+        cutout_obs = cut_square_box(obs.data, center, bx)
```

Comment:
I am testing this code on HSC data and `cut_square_box` fails for multichannel observations as it assumes 2D inputs. Am I missing something here? Or should we update this function to handle multichannel observations?

Reply:
![Screenshot 2024-05-03 at 9 20 47 AM](https://github.com/pmelchior/scarlet2/assets/44228257/93465307-0e65-4eac-8d98-4108dc271201)


This should be fixed now